pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -15,18 +15,41 @@
  */
 package com.linkedin.pinot.core.operator.filter;
 
+import com.linkedin.pinot.core.common.DataSource;
+import com.linkedin.pinot.core.common.DataSourceMetadata;
 import com.linkedin.pinot.core.operator.BaseOperator;
 import com.linkedin.pinot.core.operator.blocks.BaseFilterBlock;
-
+import com.linkedin.pinot.core.operator.filter.predicate.PredicateEvaluator;
+import javax.annotation.Nullable;
 
 /**
  * Base Operator for all filter operators. ResultBlock is initialized in the planning phase
  *
  */
 public abstract class BaseFilterOperator extends BaseOperator<BaseFilterBlock> {
 
+  protected DataSource _dataSource;
+
+  protected PredicateEvaluator _predicateEvaluator;
+
   /**
    * Return whether the result is empty.
    */
   public abstract boolean isResultEmpty();
+
+  /**
+   * Returns the datasource metadata
+   */
+  public @Nullable DataSourceMetadata getDataSourceMetadata() {
```

Comment:
can we expose columnmetadata instead of datasourcemetadata. We dont want to expose datasourcemetadata outside of datasource.

Reply:
Will address this in a followup PR. Thanks!